### PR TITLE
DROOLS-5848 : CallMethod with ActionFieldFunctions does not compile

### PR DESCRIPTION
**Pull request by Rikkola**
**Thank you for submitting this pull request**

**JIRA**: 

[link](https://issues.redhat.com/browse/DROOLS-5848)
 

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by Rikkola** [comment]
Jenkins retest this

**Comment by Rikkola** [comment]
@jstastny-cz @danielzhe Would you two be kind enough to review?

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java [hunk 1]
```diff
@@ -1227,6 +1228,52 @@ public void testActionCallIsNotIgnored() {
...
+        final ActionCallMethod actionCallMethod = new ActionCallMethod("x");
+        actionCallMethod.setState(ActionCallMethod.TYPE_DEFINED);
+        actionCallMethod.setMethodName("add");
+        actionCallMethod.addFieldValue(new ActionFieldFunction("name",
+                                                               "Toni",
+                                                               DataType.TYPE_STRING));
+
+        brlAction1.getDefinition().add(actionCallMethod);
+        brlAction1.getChildColumns().add(new BRLActionVariableColumn("",
+                                                                     DataType.TYPE_BOOLEAN,
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java:1257]: Shouldn't we use `DataType.TYPE_VOID`? Or isn't this the return type of the method `add` ?

**Review by danielzhe — APPROVED** [review]

**Review by jomarko — CHANGES_REQUESTED** [review]
Sorry, I can not approve the PR. The added test will pass even if I revert `GuidedDTDRLPersistence` changes. I tried also to debug the added test and put breakpoint to the `GuidedDTDRLPersistence`changes. The changed private method was never reached.

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java [hunk 2]
```diff
@@ -1227,6 +1228,58 @@ public void testActionCallIsNotIgnored() {
...
+        final String drl = GuidedDTDRLPersistence.getInstance().marshal(dt);
+        final String expected = "//from row number: 1\n" +
+                "//desc\n" +
+                "rule \"Row 1 null\"\n" +
+                "dialect \"mvel\"\n" +
+                "when\n" +
+                "  x : Context( )\n" +
+                "then\n" +
+                "  x.add( \"Toni\" );\n" +
+                "  x.setName( \"ToniR\" );\n" +
```
Comment (Rikkola) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java:1279]: @jomarko The problem with the old fix was that this line was never added to the DRL.
The problem appears when there is `ActionCallMethod` in a BRL before Set or Update column that is not BRL. The two columns use the same fact variable, so when searching by name the `ActionCallMethod` is found and we try to append the set/update to that. We should just ignore the call.

**Review by jomarko — APPROVED** [review]
@Rikkola thank you for the PR. Now it seems to me as proper reproducer. I tried your test on master and it throws the same exception as attached in jira. Before merge I would like to add one more test [1] as you mention also update in https://github.com/kiegroup/drools/pull/3263/files#r535945210 

[1]
https://github.com/Rikkola/drools/pull/8

**Comment by Rikkola** [comment]
jenkins retest this

**Comment by Rikkola** [comment]
jenkins retest this

**Comment by Rikkola** [comment]
jenkins retest this

**Comment by Rikkola** [comment]
jenkins retest this

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
org.kie.server.integrationtests.drools.pmml.ApplyScorecardModelIntegrationTest.testApplyPmmlScorecardInBatch[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:42511/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.pmml.ApplyScorecardModelIntegrationTest.testApplyPmmlScorecard[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:42511/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.pmml.ApplyScorecardMultiModuleIntegrationTest.buildAndDeployTest[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:42511/kie-server-services/services/rest/server'}]

Jenkins run fdb

**Comment by mareknovotny** [comment]
jenkins do cdb

**Comment by Rikkola** [comment]
jenkins retest this

**Comment by Rikkola** [comment]
org.kie.server.integrationtests.drools.pmml.ApplyScorecardModelIntegrationTest.testApplyPmmlScorecardInBatch[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:35266/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.pmml.ApplyScorecardModelIntegrationTest.testApplyPmmlScorecard[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:35266/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.pmml.ApplyScorecardMultiModuleIntegrationTest.buildAndDeployTest[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:35266/kie-server-services/services/rest/server'}]

Jenkins run fdb

**Comment by Rikkola** [comment]
[2020-12-16T11:10:04.738Z] stderr: fatal: unable to access 'https://github.com/kiegroup/kie-soup.git/': Failed connect to github.com:443; Connection timed out

Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Local test run in kie-wb-distributions passed.

Jenkins run fdb

**Comment by mareknovotny** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by Ginxo** [comment]
Jenkins do fdb

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers QA changed to jomarko. So we have both QA and technical reviews. Can I get a merge please.

**Pull request merged**